src/Goetas/Twital/Attribute/AttrAttribute.php

```diff
@@ -13,9 +14,9 @@
  */
 class AttrAttribute implements Attribute
 {
-    public static function getVarname(\DOMNode $node)
+    public static function getVarname(\DOMElement $node)
     {
-        return "__a9" . strtr(md5($node->ownerDocument->saveXML($node)).spl_object_hash($node), "-","_");
+        return "__a9" . strtr($node->getAttributeNS(Twital::NS, '__internal-id__').spl_object_hash($node), "-","_");
```

Comment:
Oh, I get it now. First part of hash is unique for current document and second part ensures that it won't colide with normal variables.